src/backend.rs

```diff
@@ -186,3 +186,127 @@ mod file {
         }
     }
 }
+
+
+#[cfg(feature = "file-path-backend")]
+pub use self::file_path::FilePath;
+
+#[cfg(feature = "file-path-backend")]
+mod file_path {
+    use std::path::PathBuf;
+
+    use async_trait::async_trait;
+    use tokio::io::{AsyncReadExt, AsyncWriteExt};
+
+    use super::Backend;
+
+    /// A file-path-backed backend. It does not keep the database file open and only access it when needed.
+    /// 
+    /// Saving is first done to a temporary file (with `.tmp` extension appended),
+    /// which is renamed over the real file in case of success.
+    /// 
+    /// You may want to `fsync` the parent directory after saving to ensure the data is actually persisted to disk.
+    /// 
+    /// Note that this implementation is not protected against symlink shenanigans that can redirect the file write elsewhere.
+    ///
+    /// Note: this requires its futures to be executed on the Tokio 0.3 runtime.
+    #[cfg_attr(docsrs, doc(cfg(feature = "file-backend")))]
+    #[derive(Debug)]
+    pub struct FilePath(PathBuf);
+
+    impl FilePath {
+        /// Creates the backend by ensuring specified file can be opened for reading and writing.
+        ///
+        /// # Errors
+        ///
+        /// If the file does not exist or could not be opened for reading and writing, an error
+        /// variant is returned.
+        pub async fn from_path(path: PathBuf) -> Result<Self, std::io::Error>
+        {
+            let f = tokio::fs::OpenOptions::new()
+                    .read(true)
+                    .write(true)
+                    .open(&path)
+                    .await?;
+            
+            // Close the file, as we don't need it yet and opened it only as a check.
+            drop(f);
```

Comment:
This drop is not necessary, as the file will be closed once this function returns.

Having a comment that the file is only opened for checking is still useful.

Reply:
I think explicit `drop`s are OK when dropping is significant (especially if there is comment speaking about the drop).

If for some reason more code is added before the final return, one would need to stop and think whether to add it before or after the `drop`.

Also if the type gets changed to a reference, the drop starts emitting warning like "dropping a reference is a no-op", telling that the designated dropping place is no longer where the drop actually happens.